File: BFS And DFS ALgorithm/Adjacency_List.cpp
#include <bits/stdc++.h>
using namespace std;
class Node
{
public:
    int data;
    Node *next;
    Node(int val)
    {
        data = val;
        next = nullptr;
    }
};
class Linked_l
{
private:
    Node *head;

public:
    Linked_l()
    {
        head = nullptr;
    }
    void Push(int val)
    {
        if (head == NULL)
        {
            head=new Node(val);
            return;
        }
        Node *temp = head;
        Node *newNode = new Node(val);
        while (temp->next != NULL)
        {
            temp = temp->next;
        }
        temp->next = newNode;
    }
    void display()
    {
        Node *temp = head;
        if(temp!=nullptr){
            cout <<temp->data << "->";
            temp=temp->next;
        }
        while (temp != nullptr)
        {
            cout << temp->data;
            if(temp->next!=nullptr){
                cout<<",";
            }
            temp = temp->next;
        }
        cout << endl;
    }
};
class graph
{
    public:
    void addedge(int x, int y,Linked_l l[])
    {
        l[x].Push(y);
        l[y].Push(x);
    }
    void display(Linked_l l[],int vertices)
    {
        for (int i = 0; i < vertices; i++)
        {
            l[i].display();
        }
    }
};
int main()
{
    int e,v;
    cout<<"Enter Total Number Of Vertices In Graph \n-> ";
    cin>>v;
    cout<<"Enter Total Number Of Edges In Graph \n->";
    cin>>e;
    graph g;
    Linked_l *l=new Linked_l[v];
    for(int i=0;i<v;i++){
        l[i].Push(i);
    }
    for(int i=0;i<e;i++){
        int v1,v2;
        cout<<"\nEdge Number "<<i+1<<endl;
        cout<<"Vertex :";
        cin>>v1;
        cout<<"Vertex :";
        cin>>v2;
        g.addedge(v1,v2,l);
    }
    cout<<"\n\nAdjacency List is As Follows "<<endl;
    g.display(l,v);
    return 0;
}

File: Prim's Algorithm/Adjacency_List.java
public class Adjacency_List {
    static class Node{
        int vertices;
        int weight;
        Node next;
        public Node(int val,int w){
            vertices=val;
            weight=w;
            next=null;
        }
    }
    static class Linked_lost{
        Node head;
        public Linked_lost(){
            head=null;
        }
        public void Push(int val,int w){
            Node temp=head;
            Node newNode=new Node(val,w);
            if(head==null){
                head=new Node(val,w);
                System.out.println(head.vertices);
                return;
            }
            while(temp.next!=null){
                temp=temp.next;
            }
            temp.next=newNode;
            return;
        }
        public void isempty(){
            if(head==null){
                System.out.println("List Is empty ");
                return;
            }
        }
        public void display(){
            Node temp=head;
            while(temp!=null){
                System.out.print(temp.vertices+" ");
                temp=temp.next;
            }
            System.out.println();
        }
    }
    static class Graph{
        int vertices;
        Linked_lost l[];
        public Graph(int v){
            vertices=v;
            l=new Linked_lost[vertices];
            for(int i=0;i<vertices;i++){
                l[i]=new Linked_lost();
            }
        }
        void addedge(int m,int n,int w){
            l[m].Push(n,w);
            l[n].Push(m,w);
        }
        void display(){
            for(int i=0;i<vertices;i++){
                System.out.print(i+"->");
                l[i].display();
            }
        }
    }
    public static void main(String[] args) {
        Graph g=new Graph(4);
        g.addedge(0, 1, 1);
        g.addedge(0, 2, 4);
        g.addedge(1, 2, 2);
        g.addedge(2, 3, 3);
        g.addedge(3, 1, 4);
        g.display();
    }
}
